### DISCUSSION TITLE
Why hasn't Relm4 taken off?

### DISCUSSION THREAD

**Discussion by sonnyp** [Relm4/Relm4#555] [General]
Hello,

GNOME contributor here. I'm interested in understanding why reactive GTK hasn't taken off.
Particularly in the context of GNOME but not only.

https://floss.social/@sonny/111233824756696075

I would be curious and grateful to hear your thoughts.

**Comment by nikableh** [comment]
Honestly, I think it's not even about if it's reactive or not. Visuals/looks, ease of use and maturity (in that order) tend to be the most important factors. And when you comes to choosing the framework of an app people have many options to choose from:

- What framework? (Qt? GTK? Electron?)
- Why subset of a framework? (Qt Widget? QML? GNOME HIG? Default GTK?)
- Do I want to make native app or do I wrap it in Electron?

On each of these steps there is an option to narrow yourself to something specific. And then there is a fact that not a lot of people make desktop applications in the first place. A lot of UI nowadays is web or something that wraps web to be used as if it was actually native (like Electron).

I don't think desktop applications are going anywhere, it's just that there are fewer people that make them.

No matter how much hate GNOME gets, its HIG is very well thought-out and there'll always be people who would like to make apps with GTK+Adwaita.
